### PR TITLE
Bump GCT, propagate context information

### PR DESCRIPTION
**Pull request by lrascao**

### CODE REVIEW

**Comment by ydm** [comment]
No. :)

1. this PR just fixes compilation
2. breaks API and complicates it absolutely without a real need
3. does not take advantage of the powerful concept that is the Context
4. context is also the key for a graceful exit. if we merge this PR, it will be either impossible or harder to implement

**Comment by lrascao** [comment]
> No. :)
> 
> 1. this PR just fixes compilation

yes, this is the main motivation, GCT recently merged this to master so we need to unblock the upgrade path, there are/will be other improvements to GCT that we'll want to take advantage of and we don't want to be keeping a fork of GCT

> 2. breaks API and complicates it absolutely without a good reason

not a problem as there is no Dola release so far, no complication whatsoever, context is an idiomatic and expected parameter in a lot of APIs

> 3. does not take advantage of the powerful concept that is the Context

That is beyond the scope of this PR, full usage of context is to be tackled on a separate task, after all it wasn't being used so far and continues so, now it's just a little bit easier to start using it.

> 4. context is also the key to a graceful exit. if we merge this PR, it will be either impossible or harder to implement

I'm not getting why that is, context seems like something that would come in handy when dealing with graceful exit.

File: strategy_balances.go [hunk 1]
```diff
@@ -88,7 +89,7 @@ func (b *BalancesStrategy) Currency(exchangeName string, code string, accountID
 }
 
 func (b *BalancesStrategy) tick(k *Keep, e exchange.IBotExchange) {
-	holdings, err := e.UpdateAccountInfo(asset.Spot)
+	holdings, err := e.UpdateAccountInfo(context.Background(), asset.Spot)
```
Comment (ydm) [inline strategy_balances.go:92]: can u copy-paste the KeepBuilder part from the ctx branch and use k.ctx here?

**Comment by lrascao** [comment]
Played around a bit in `ebstrategy` using this branch, the normal scenario is using `keep.Ctx()` and this allows us to cancel in-flight requests on exit. I then tried the `WithTimeout` scenario by enforcing a max timeout on the `FetchTradablePairs` request and it functioned as expected

**Pull request merged**